lib/archethic/contracts/interpreter/common_interpreter.ex

```diff
@@ -343,23 +339,29 @@ defmodule Archethic.Contracts.Interpreter.CommonInterpreter do
     {new_node, acc}
   end
 
-  def postwalk(node = {{:atom, function_name}, _, args}, acc, function_keys) when is_list(args) do
-    if Enum.member?(function_keys, {function_name, length(args)}) do
-      new_node =
-        quote do
-          Scope.execute_function_ast(unquote(function_name), unquote(args))
-        end
+  def postwalk(node = {{:atom, function_name}, _, args}, acc = %{functions: functions})
+      when is_list(args) do
+    arity = length(args)
 
-      {new_node, acc}
-    else
-      reason = "The function #{function_name}/#{Integer.to_string(length(args))} does not exist"
+    case Enum.find(functions, fn
```

Comment:
maybe the `defp function_exists?(functions, function_name, arity) do` from conditionInterpreter shall be extracted and re-used?